### PR TITLE
Report warning on toc folder-inclusion not found

### PR DESCRIPTION
**Pull request by mingwli**
Fixing [AB#206730](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/206730)
image

- Report warning when no toc.yml nor toc.md found using folder relative syntax to include sub-toc file.
- Move file-not-found under Link category

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5805)

### CODE REVIEW

File: src/docfx/build/toc/TableOfContentsLoader.cs [hunk 1]
```diff
@@ -333,6 +333,10 @@ private IReadOnlyList<string> GetMonikers(TableOfContentsNode currentItem, List<
                             }
                         }
                     }
+                    if (result == null)
+                    {
+                        errors.Add(Errors.Link.FileNotFound($"Unable to find either toc.yml or toc.md inside {href}/. Please make sure the file exists", href));
```
Comment (herohua) [inline src/docfx/build/toc/TableOfContentsLoader.cs:338]: don't append `/`

File: docs/specs/toc.yml [hunk 2]
```diff
@@ -707,6 +707,7 @@ outputs:
     {"items":[{"name":"Duplicated keys","href":"f1/b"}]}
   .errors.log: |
     ["suggestion","yaml-duplicate-key","Key 'topicHref' is already defined, remove the duplicate key. NOTE: This Suggestion will become a Warning on 06/30/2020.","docs/TOC.yml",3,3]
+    ["warning","file-not-found","Unable to find either toc.yml or toc.md inside f1//. Please make sure the file exists","docs/TOC.yml",4,12]
```
Comment (herohua) [inline docs/specs/toc.yml:710]: how about just printing `... inside f1/`?

File: src/docfx/build/toc/TableOfContentsLoader.cs (hunk 1 above)
Comment (yufeih) [inline src/docfx/build/toc/TableOfContentsLoader.cs:338]: Prefer putting message text inside `Error.cs`

File: docs/specs/toc.yml [hunk 3]
```diff
@@ -707,6 +707,7 @@ outputs:
     {"items":[{"name":"Duplicated keys","href":"f1/b"}]}
   .errors.log: |
     ["suggestion","yaml-duplicate-key","Key 'topicHref' is already defined, remove the duplicate key. NOTE: This Suggestion will become a Warning on 06/30/2020.","docs/TOC.yml",3,3]
+    ["warning","file-not-found","Unable to find either toc.yml or toc.md inside f1/ Please make sure the file exists","docs/TOC.yml",4,12]
```
Comment (herohua) [inline docs/specs/toc.yml:710]: better to say `in folder f1`

File: docs/specs/legacy.yml [hunk 4]
```diff
@@ -390,7 +390,6 @@ outputs:
     }
 ---
 # Orphan file, with multiple toc of same site path, look up by file path(sub then parent)
-os: windows
```
Comment (live1206) [inline docs/specs/legacy.yml:393]: why is this working on all platforms now?

Comment (mingwli) [reply docs/specs/legacy.yml:393]: Test-case's been updated

**Pull request merged**